File: java/src/main/java/core/Pipe.java
package core;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.SparkFiles;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class Pipe {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("Pipe");
        JavaSparkContext sc = new JavaSparkContext(conf);

        String columnCountScript = "./src/main/scripts/columncount.py";
        String columnCountScriptName = "columncount.py";
        sc.addFile(columnCountScript);

        JavaRDD<String> lines = sc.parallelize(Arrays.asList("1,2,3", "4,5", "6", "7,8,9,10"));
        List<String> columnCounts = lines.pipe(SparkFiles.get(columnCountScriptName)).collect();

        System.out.println(columnCounts);

        sc.stop();
    }
}

File: scala/src/main/scala/core/Pipe.scala
package core

import org.apache.spark.{SparkConf, SparkContext, SparkFiles}

object Pipe {

  def main(args: Array[String]) {
    run(args)
  }

  def run(args: Array[String]) {
    val conf = new SparkConf().setAppName("Pipe")
    val sc = new SparkContext(conf)

    val columnCountScript = "./src/main/scripts/columncount.py"
    val columnCountScriptName = "columncount.py"
    sc.addFile(columnCountScript)

    val lines = sc.parallelize(Array("1,2,3", "4,5", "6", "7,8,9,10"))

    lines.pipe(SparkFiles.get(columnCountScriptName)).foreach(println)

    sc.stop()
  }
}
